crowbar_framework/app/models/service_object.rb

```diff
@@ -854,6 +854,56 @@ def violates_admin_constraint?(elements, role, nodes_is_admin = {})
     false
   end
 
+  def violates_platform_constraint?(elements, role)
+    if role_constraints[role]
```

Comment:
Please add a check that the constraint is present: ``&& role_constraints[role].has_key?("platform")``